snuba/utils/bucket_timer.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from collections import defaultdict, deque
+from datetime import datetime, timedelta
+from typing import Deque, List, Optional
+
+from snuba import environment
+from snuba.utils.metrics.wrapper import MetricsWrapper
+
+WINDOW_SIZE = timedelta(minutes=10)  # minutes
+
+
+class Bucket:
```

Comment:
It seems this could be a dataclass (from dataclasses import dataclass) . So you will not have to write the constructor.